### PR TITLE
#1254 Support VB.NET's case-insensitive assembly attributes

### PR DESCRIPTION
**Pull request by forgetaboutit**
Use case insensitive regex matching when looking for assembly attributes in `.vb` assembly info files.

Fixes issue #1254.

### CODE REVIEW

File: src/app/FakeLib/AssemblyInfoFile.fs [hunk 1]
```diff
@@ -14,6 +14,13 @@ let private regexAttrNameValueFs = @"^\s*\[\<\s*assembly:\s*(?<name>\w+?)\s*\((?
 let private regexAttrNameValueCpp = @"^\s*\[\s*assembly:\s*(?<name>\w+?)\s*\((?<value>.*)\)\s*\]\s*;\s*$"
 let private regexAttrNameValueVb = @"^\s*\<\s*assembly:\s*(?<name>\w+?)\s*\((?<value>.*)\)\s*\>\s*$"
 
+// additional regex options for matching assembly attributes
+let private regexAttrOptionsCs = Option.None
+let private regexAttrOptionsFs = Option.None
+let private regexAttrOptionsCpp = Option.None
+// VB.NET is case-insensitive
+let private regexAttrOptionsVb = Option.Some (RegexOptions.IgnoreCase)
+
```
Comment (inosik) [inline src/app/FakeLib/AssemblyInfoFile.fs]: I don't think we need these.

File: src/app/FakeLib/AssemblyInfoFile.fs [hunk 2]
```diff
@@ -284,15 +291,22 @@ let private removeAtEnd (textToRemove:string) (text:string) =
...
-    let regex = 
-        if assemblyInfoFile.ToLower().EndsWith(".cs") then regexAttrNameValueCs
-        elif assemblyInfoFile.ToLower().EndsWith(".fs") then regexAttrNameValueFs
-        elif assemblyInfoFile.ToLower().EndsWith(".vb") then regexAttrNameValueVb
-        elif assemblyInfoFile.ToLower().EndsWith(".cpp") then regexAttrNameValueCpp
+    let (regex, additionalOptions) =
+        if assemblyInfoFile.ToLower().EndsWith(".cs") then (regexAttrNameValueCs, regexAttrOptionsCs)
+        elif assemblyInfoFile.ToLower().EndsWith(".fs") then (regexAttrNameValueFs, regexAttrOptionsFs)
+        elif assemblyInfoFile.ToLower().EndsWith(".vb") then (regexAttrNameValueVb, regexAttrOptionsVb)
+        elif assemblyInfoFile.ToLower().EndsWith(".cpp") then (regexAttrNameValueCpp, regexAttrOptionsCpp)
```
Comment (inosik) [inline src/app/FakeLib/AssemblyInfoFile.fs]: Simply return `RegexOptions.IgnoreCase` for VB and `RegexOptions.None` for everything else.

Comment (forgetaboutit) [inline src/app/FakeLib/AssemblyInfoFile.fs]: You're right. I will fix this.

**Comment by forgetaboutit** [comment]
I fixed the issues you pointed out and updated the commit accordingly.

**Comment by inosik** [comment]
:+1:

Maybe you should also extend the specs [here](https://github.com/fsharp/FAKE/blob/60acb2f035cb39f2bd40a6c58a4de10d915da248/src/test/Test.FAKECore/AssemblyInfoSpecs.cs#L265)?

**Comment by forgetaboutit** [comment]
Digging into the code a bit more, I discovered the `UpdateAttributes` function, which is also exhibits the same problem. I will fix this as well and extend the specs accordingly. Please wait with the merge until I'm done.

**Comment by forgetaboutit** [comment]
The new commit addresses both issues in `GetAttributes` and `UpdateAttributes` and extends the specs accordingly. I can think of nothing else that would prevent merging the fixes, but I'd be happy to fix any issues.

**Comment by forki** [comment]
Awesome. thanks for taking care of this

**Pull request merged**